beam/core-testing/src/main/java/cz/o2/proxima/beam/storage/TestStreamStorage.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.beam.storage;
+
+import com.google.common.base.Preconditions;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import cz.o2.proxima.beam.core.BeamDataOperator;
+import cz.o2.proxima.beam.core.DataAccessor;
+import cz.o2.proxima.beam.core.DataAccessorFactory;
+import cz.o2.proxima.repository.AttributeDescriptor;
+import cz.o2.proxima.repository.AttributeFamilyDescriptor;
+import cz.o2.proxima.repository.EntityDescriptor;
+import cz.o2.proxima.repository.Repository;
+import cz.o2.proxima.storage.StreamElement;
+import cz.o2.proxima.storage.commitlog.Position;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.testing.TestStream;
+import org.apache.beam.sdk.values.PCollection;
+
+/**
+ * Storage with URI scheme {@code test-stream://}. This storage is backed by {@link TestStream} and
+ * enables fine tuning with regard to watermark and other stream aspects.
+ */
+public class TestStreamStorage implements DataAccessorFactory {
+
+  private static final Map<Repository, StreamProviders> storages = new ConcurrentHashMap<>();
+
+  private static class StreamProviders {
+    private final Map<URI, TestStream<StreamElement>> streams = new HashMap<>();
+
+    TestStream<StreamElement> get(URI uri) {
+      return Objects.requireNonNull(streams.get(uri));
+    }
+
+    void put(URI uri, TestStream<StreamElement> stream) {
+      streams.put(uri, stream);
+    }
+  }
+
+  /**
+   * Replace all storages in all attribute families in given config with {@code test-stream}.
+   *
+   * @param config the configuration to replace
+   * @return replaced config
+   */
+  public static Config replaceStorages(Config config) {
+    final Map<String, String> overrides = new HashMap<>();
+    config
+        .getObject("attributeFamilies")
```

Comment:
Can we move this as constant into `ConfigConstants`?

Reply:
Already did in parallel. :-)